### PR TITLE
MAINT: Report download time and size

### PR DESCRIPTION
**Pull request by larsoner**
CircleCI has timed out a couple of times lately due to our 3h limit. It costs more $$$ to increase this limit and I think it's pretty reasonable anyway, so let's try to make it faster. 2h is spent in the doc build mostly due to examples/tutorials -- which we should certainly continue to try to optimize -- but recently 45min has been spent in dataset downloading. This PR adds some diagnostic download time/size `logger` to our fetching that should help us see it in logs, and might be nice for users.

### CODE REVIEW

**Comment by larsoner** [comment]
Looks like this might be one culprit, looking locally after watching CircleCI have a 10-minute timeout trying to download it:
```
hf_sef_evoked.tar.gz
https://zenodo.org/record/3523071/files/hf_sef_evoked.tar.gz
731 KB/s - 124 MB of 731 MB, 14 mins left
```
we already have a mirror on `osf.io` so I'll update the file there and push

**Comment by larsoner** [comment]
Okay @drammock this one is ready for review. Mirroring data on OSF.io took time from ~45 min down to ~15 min as you can see here (along with all timings for downloads -- I killed the build once that step was done):

https://app.circleci.com/pipelines/github/mne-tools/mne-python/18689/workflows/29272f16-bc51-4ce8-ae40-c7ef59c3795f/jobs/53962

File: mne/datasets/eegbci/eegbci.py [hunk 1]
```diff
@@ -88,7 +91,8 @@ def data_path(url, path=None, force_update=False, update_path=None, *,
             # URL to one of Pooch's test files
```
Comment (drammock) [inline mne/datasets/eegbci/eegbci.py:91]: assuming this is copy-paste cruft from a pooch example script?

```suggestion
```

File: mne/datasets/limo/limo.py [hunk 2]
```diff
@@ -168,14 +171,23 @@ def data_path(subject, path=None, force_update=False, update_path=None, *,
     # use our logger level for pooch's logger too
     pooch.get_logger().setLevel(logger.getEffectiveLevel())
     # fetch the data
+    sz = 0
     for fname in ('LIMO.mat', 'Yr.mat'):
-        destination = op.join(subject_path, fname)
-        if force_update and op.isfile(destination):
-            os.remove(destination)
+        destination = Path(op.join(subject_path, fname))
```
Comment (drammock) [inline mne/datasets/limo/limo.py:176]:
```suggestion
        destination = Path(subject_path, fname)
```

File: mne/datasets/eegbci/eegbci.py [hunk 3]
```diff
@@ -195,14 +200,23 @@ def load_data(subject, runs, path=None, force_update=False, update_path=None,
 
     # fetch the file(s)
     data_paths = []
+    sz = 0
     for run in runs:
         file_part = f'S{subject:03d}/S{subject:03d}R{run:02d}.edf'
-        destination = op.join(base_path, file_part)
-        if force_update and op.isfile(destination):
-            os.remove(destination)
+        destination = Path(op.join(base_path, file_part))
```
Comment (drammock) [inline mne/datasets/eegbci/eegbci.py:206]:
```suggestion
        destination = Path(base_path, file_part)
```

File: mne/datasets/utils.py [hunk 4]
```diff
@@ -299,49 +301,52 @@ def _download_all_example_data(verbose=True):
...
+                 'brainstorm.bst_resting brainstorm.bst_phantom_ctf '
+                 'brainstorm.bst_phantom_elekta'
+                 ).split():
+        mod = importlib.import_module(f'mne.datasets.{kind}')
+        data_path_func = getattr(mod, 'data_path')
+        kwargs = dict()
+        if 'accept' in inspect.getfullargspec(data_path_func).args:
+            kwargs['accept'] = True
+        paths[kind] = data_path_func(**kwargs)
+        logger.info(f'[done {kind}]')
```
Comment (drammock) [inline mne/datasets/utils.py:318]: 😍

**Review by drammock — COMMENTED** [review]
this is a nice improvement. CIs are not happy though. Cirrus fails with

```
mne/datasets/sleep_physionet/tests/test_physionet.py:132: in test_sleep_physionet_age
    paths = age.fetch_data(subjects=[0], recording=[1], path=physionet_tmpdir)
<decorator-gen-588>:12: in fetch_data
    ???
mne/datasets/sleep_physionet/age.py:129: in fetch_data
    sz += os.path.getsize(psg_fname)
/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/genericpath.py:50: in getsize
    return os.stat(filename).st_size
E   FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/76/zy5ktkns50v6gt5g8r0sf6sc0000gn/T/pytest-of-admin/pytest-0/physionet_files0/physionet-sleep-data/SC4001E0-PSG.edf'
```

which looks related / legit; haven't looked at any other failures

**Comment by drammock** [comment]
there are also 

```
 _______________________ test_manifest_check_download[1] ________________________
mne/datasets/tests/test_datasets.py:225: in test_manifest_check_download
    _manifest_check_download(manifest_path, destination, url, hash_)
mne/datasets/utils.py:570: in _manifest_check_download
    pooch.retrieve(
E   TypeError: _fake_zip_fetch() got an unexpected keyword argument 'downloader'
```

**Comment by drammock** [comment]
I think this PR broke circle:

```
WARNING: /home/circleci/project/tutorials/preprocessing/40_artifact_correction_ica.py failed to execute correctly: Traceback (most recent call last):
  File "/home/circleci/project/tutorials/preprocessing/40_artifact_correction_ica.py", line 539, in <module>
    fname = mne.datasets.eegbci.load_data(subj + 1, runs=[3])[0]
IndexError: list index out of range
```

**Pull request merged**